app/src/main/java/com/sjh/sunflower_sksowk156/core/common/result/DataStatus.kt

```diff
@@ -0,0 +1,16 @@
+package com.sjh.sunflower_sksowk156.core.common.result
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onStart
+
+sealed interface DataStatus<out T> {
+    data class Success<T>(val data: T) : DataStatus<T>
+    data class Error(val exception: Throwable) : DataStatus<Nothing>
+    data object Loading : DataStatus<Nothing>
+}
+
+fun <T> Flow<T>.toSafeFlow(): Flow<DataStatus<T>> =
+    map<T, DataStatus<T>> { DataStatus.Success(it) }.onStart { emit(DataStatus.Loading) }
+        .catch { emit(DataStatus.Error(it)) }
```

Comment:
```suggestion
fun <T> Flow<T>.toSafeFlow(): Flow<DataStatus<T>> =
    map<T, DataStatus<T>>(DataStatus::Success).onStart { emit(DataStatus.Loading) }
        .catch { emit(DataStatus.Error(it)) }
```
function reference를 이용하면 이렇게 적을 수 있습니다.